spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageConverterConfigurer.java

```diff
@@ -83,25 +88,35 @@
 	private final BindingServiceProperties bindingServiceProperties;
 
 	private ConfigurableListableBeanFactory beanFactory;
+	
+	private final Map<String, PartitionKeyExtractorStrategy> partitionKeyExtractors;
+	
+	private final Map<String, PartitionSelectorStrategy> partitionSelectors;
 
+	@SuppressWarnings("unchecked")
 	public MessageConverterConfigurer(BindingServiceProperties bindingServiceProperties,
 			CompositeMessageConverterFactory compositeMessageConverterFactory) {
+		this(bindingServiceProperties, compositeMessageConverterFactory, Collections.EMPTY_MAP, Collections.EMPTY_MAP);
+	}
+	
+	@SuppressWarnings("unchecked")
+	public MessageConverterConfigurer(BindingServiceProperties bindingServiceProperties,
+			CompositeMessageConverterFactory compositeMessageConverterFactory,
+			Map<String, PartitionKeyExtractorStrategy> partitionKeyExtractors,
+			Map<String, PartitionSelectorStrategy> partitionSelectors) {
 		Assert.notNull(compositeMessageConverterFactory,
 				"The message converter factory cannot be null");
 		this.bindingServiceProperties = bindingServiceProperties;
 		this.compositeMessageConverterFactory = compositeMessageConverterFactory;
+		this.partitionKeyExtractors = partitionKeyExtractors == null ? Collections.EMPTY_MAP : partitionKeyExtractors;
+		this.partitionSelectors = partitionSelectors == null ? Collections.EMPTY_MAP : partitionSelectors;
 	}
 
 	@Override
 	public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 		this.beanFactory = (ConfigurableListableBeanFactory) beanFactory;
 	}
 
-	@Override
-	public void afterPropertiesSet() throws Exception {
-		Assert.notNull(this.beanFactory, "Bean factory cannot be empty");
```

Comment:
I think we still need this, otherwise we'll get this log warning...

    logger.warn("Creating EvaluationContext with no beanFactory", new RuntimeException("No beanFactory"));

from `ExpressionUtils`.

Reply:
Hmm, but this was just validating that it is there and in our case due to BFAware it always is since its a bean. Right?

Reply:
True, but only as long as nobody instantiates it outside of Spring for some reason.

Reply:
And I think I am ok with that since outside of Spring this method may not even execute (unless someone calls it explicitly)